src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -92,114 +95,273 @@ private async Task GetProjectMetadataAsync(
         {
-            List<PackageSource> httpPackageSources = null;
-            foreach (PackageSource packageSource in listPackageArgs.PackageSources)
+            foreach (var package in packages)
             {
-                if (packageSource.IsHttp && !packageSource.IsHttps && !packageSource.AllowInsecureConnections)
+                var vuln = GetPackageVulnerabilities(
+                    vulnerabilities,
+                    package.Name,
+                    package.ResolvedPackageMetadata.Identity.Version.ToNormalizedString()
+                    ).ToList();
+
+                if (vuln != null && vuln.Count > 0)
                 {
-                    if (httpPackageSources == null)
-                    {
-                        httpPackageSources = new();
-                    }
-                    httpPackageSources.Add(packageSource);
+                    reportPackages.Add(
+                        new ListReportPackage(
+                            package.Name,
+                            package.ResolvedPackageMetadata.Identity.Version.ToString(),
+                            vuln));
                 }
             }
+        }
+
+        private static bool IsProjectAssetsFileValid(string assetsPath, string projectPath, ListPackageProjectModel projectModel, out LockFile assetsFile)
+        {
+            assetsFile = null;
 
-            if (httpPackageSources != null && httpPackageSources.Count != 0)
+            if (!File.Exists(assetsPath))
+            {
+                projectModel.AddProjectInformation(ProblemType.Error,
+                    string.Format(CultureInfo.CurrentCulture, Strings.Error_AssetsFileNotFound, projectPath));
+                return false;
+            }
+            else
             {
-                if (httpPackageSources.Count == 1)
+                var lockFileFormat = new LockFileFormat();
+                assetsFile = lockFileFormat.Read(assetsPath);
+
+                // Assets file validation
+                if (assetsFile.PackageSpec == null ||
+                    assetsFile.Targets == null ||
+                    assetsFile.Targets.Count == 0)
                 {
-                    projectModel.AddProjectInformation(
-                        ProblemType.Warning,
-                        string.Format(CultureInfo.CurrentCulture,
-                        Strings.Warning_HttpServerUsage,
-                        "list package",
-                        httpPackageSources[0]));
+                    projectModel.AddProjectInformation(ProblemType.Error,
+                        string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_ErrorReadingAssetsFile, assetsPath));
+                    return false;
                 }
                 else
+                {
+                    return true;
+                }
+            }
+        }
+
+        private static bool IsProjectPackageReference(Project project, ListPackageProjectModel projectModel, string projectPath)
```

Comment:
fwiw, I feel like this refactoring made the code longer compared to what it was previously. 

The previous method was well named, and this method is really just forwarding the arguments and return true/false. 
Doesn't add much.